lib/src/main/java/com/auth0/spring/security/api/authentication/AuthenticationJsonWebToken.java

```diff
@@ -83,4 +82,34 @@ public void setAuthenticated(boolean isAuthenticated) throws IllegalArgumentExce
     public String getName() {
         return decoded.getSubject();
     }
+
+    private List<SimpleGrantedAuthority> getScopeAuthorities() {
+        String scope = decoded.getClaim("scope").asString();
+        if (scope == null || scope.trim().isEmpty()) {
+            return Collections.emptyList();
+        }
+        final String[] scopes = scope.split(" ");
+        List<SimpleGrantedAuthority> authorities = new ArrayList<>(scopes.length * 2);
+        for (String value : scopes) {
+            // For backwards-compatibility, create authority without scope prefix
+            authorities.add(new SimpleGrantedAuthority(value));
+            authorities.add(new SimpleGrantedAuthority(SCOPE_AUTHORITY_PREFIX + value));
+        }
+        return authorities;
+    }
+
+    private List<SimpleGrantedAuthority> getPermissionAuthorities() {
+        String[] permissions = decoded.getClaim("permissions").asArray(String.class);
```

Comment:
are we 100% sure, for auth0 use case at least, that this is always an array of strings? Could it happen to be a JSON object instead?

Reply:
When defining permissions for an API in the dashboard, every value becomes a string in the token.